pinot-common/src/main/java/org/apache/pinot/common/utils/ServiceStatus.java

```diff
@@ -189,20 +194,50 @@ public IdealStateMatchServiceStatusCallback(HelixManager helixManager, String cl
 
     protected abstract String getMatchName();
 
+    private boolean isDone() {
+      if (_resourcesToMonitor.isEmpty() || _numTotalResourcesToMonitor <= 0) {
+        return true;
+      }
+      if ((double)(_numTotalResourcesToMonitor - _resourcesToMonitor.size())*100/_numTotalResourcesToMonitor
```

Comment:
We can estimate the number of resources we need to be "done" _once_ and compare against it.

Reply:
done